src/ZODB/Connection.py

```diff
@@ -302,8 +302,8 @@ def close(self, primary=True):
                     f()
                 except: # except what?
                     f = getattr(f, 'im_self', f)
-                    self._log.error("Close callback failed for %s", f,
-                                    exc_info=sys.exc_info())
+                    self._log.error("Close callback failed for %s %s", f,
+                                    sys.exc_info())
             self.__onCloseCallbacks = None
```

Comment:
So this is ok in its original form because it's an error, whereas the change below was breaking because it was a warning. Got it.